lib/array.js

```diff
@@ -126,10 +130,27 @@ internals.checkItems = function (items, wasArray, state, options) {
             continue;
         }
 
+        // Requireds
+
+        for (i = 0, il = requireds.length; i < il; ++i) {
+            var res = requireds[i]._validate(item, localState, {});
+            if (!res.errors) {
+                items[v] = res.value;
+                isValid = true;
+                inclusions.push(requireds[i]);
+                requireds.splice(i, 1);
```

Comment:
you should probably decrease i and il after splicing, you're going to be out of bounds